modules/launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java

```diff
@@ -60,12 +61,13 @@
 
     public static void runProgram(Path sourceRootPath, Path sourcePath, boolean runServices, String[] args) {
         ProgramFile programFile;
-
         String srcPathStr = sourcePath.toString();
         if (srcPathStr.endsWith(BLangConstants.BLANG_EXEC_FILE_SUFFIX)) {
             programFile = BLangProgramLoader.read(sourcePath);
-        } else {
+        } else if (Files.isDirectory(sourcePath) || srcPathStr.endsWith(BLangConstants.BLANG_SRC_FILE_SUFFIX)) {
             programFile = compile(sourceRootPath, sourcePath);
+        } else {
+            throw new BallerinaException("Invalid File Type use .bal");
```

Comment:
The message should be something similar to "Invalid Ballerina source path, it should either be a directory or a file with a '" + BLangConstants.BLANG_SRC_FILE_SUFFIX + "' extension."